rdrf/registry/patients/admin_forms.py

```diff
@@ -212,16 +212,75 @@ def clean_signature(self):
 
 
 class PatientStageForm(forms.ModelForm):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if 'instance' in kwargs and kwargs['instance'] is not None:
+            instance = kwargs['instance']
+            stages_qs = PatientStage.objects.filter(registry=instance.registry)
+            self.fields["allowed_prev_stages"].queryset = stages_qs
+            self.fields["allowed_next_stages"].queryset = stages_qs
+            self.fields["registry"].disabled = True
+
+    def clean(self):
+        cleaneddata = self.cleaned_data
+        prev_stages = cleaneddata["allowed_prev_stages"]
+        next_stages = cleaneddata["allowed_next_stages"]
```

Comment:
we should have an

```
if self.instance:
     cleaneddata["registry"] = self.instance.registry
```

here, so that we prevent the malicious users sending in a changed `registry` value by direct POST and get around that `registry` is read-only when editing.

*Note*: at line 219 we should have `self.instance = kwargs['instance']`

Reply:
Updated.